core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -126,6 +129,21 @@ private[spark] abstract class Task[T](val stageId: Int, var partitionId: Int) ex
       taskThread.interrupt()
     }
   }
+
+  /**
+   * Deserializes the task from the broadcast variable.
+   * If Kryo serialization is being sed, a copy of the buffer is made because Kryo deserialization
+   * is not thread-safe w.r.t. the deserialization buffer (see SPARK-7708)
+   */
+  protected[this] def deserialize[T: ClassTag](taskBinary: Broadcast[Array[Byte]]): T = {
+    val ser = SparkEnv.get.closureSerializer.newInstance()
```

Comment:
How do you feel about implementing a pool in the `Task` object and accessing it from `Task`'s `run` method? This way, the serializer instance can be handed back into the pool after the task is done executing.